src/test/java/uk/ac/cam/cl/dtg/segue/api/monitors/MisuseMonitorTest.java

```diff
@@ -77,151 +80,116 @@ public final void misuseMonitorTokenOwnerLookup_checkForMisuse_emailShouldBeSent
    */
   @Test
-  public final void emailVerificationRequest_checkForMisuse_emailShouldBeSentAndExceptionShouldOccur() {
+  final void emailVerificationRequest_checkForMisuse_emailShouldBeSentAndExceptionShouldOccur() {
 
     String event = EmailVerificationRequestMisuseHandler.class.getSimpleName();
 
     IMisuseMonitor misuseMonitor = new InMemoryMisuseMonitor();
 
-    EmailVerificationRequestMisuseHandler emailVerificationMisuseHandler
-        = new EmailVerificationRequestMisuseHandler();
+    EmailVerificationRequestMisuseHandler emailVerificationMisuseHandler = new EmailVerificationRequestMisuseHandler();
 
     misuseMonitor.registerHandler(event, emailVerificationMisuseHandler);
 
-
     // Create a test user
     RegisteredUser user = new RegisteredUser();
     user.setEmail("test@test.com");
     user.setEmailVerificationStatus(EmailVerificationStatus.NOT_VERIFIED);
 
     // Soft threshold
-    try {
-      //Register the misuse monitor
-      if (misuseMonitor.hasMisused(user.getEmail(),
-          EmailVerificationRequestMisuseHandler.class.getSimpleName())) {
-        throw new SegueResourceMisuseException("Number of requests exceeded. Triggering Error Response");
-      }
-
-      for (int i = 0; i < EmailVerificationRequestMisuseHandler.SOFT_THRESHOLD; i++) {
-        misuseMonitor.notifyEvent(user.getEmail(), event);
-      }
-    } catch (SegueResourceMisuseException e) {
-      fail();
+    for (int i = 0; i < emailVerificationMisuseHandler.getSoftThreshold(); i++) {
+      assertDoesNotThrow(() -> misuseMonitor.notifyEvent(user.getEmail(), event),
+          String.format("Exception should not be thrown after %d attempts",
+              emailVerificationMisuseHandler.getSoftThreshold()));
     }
 
     // Hard threshold
-    try {
-      //Register the misuse monitor
-      if (misuseMonitor.hasMisused(user.getEmail(),
-          EmailVerificationRequestMisuseHandler.class.getSimpleName())) {
-        throw new SegueResourceMisuseException("Number of requests exceeded. Triggering Error Response");
-      }
-
-      for (int i = EmailVerificationRequestMisuseHandler.SOFT_THRESHOLD;
-           i < EmailVerificationRequestMisuseHandler.HARD_THRESHOLD; i++) {
-        misuseMonitor.notifyEvent(user.getEmail(), event);
-      }
-    } catch (SegueResourceMisuseException e) {
-      System.out.println("SegueResourceMisuseException");
+    for (int i = emailVerificationMisuseHandler.getSoftThreshold(); i < emailVerificationMisuseHandler.getHardThreshold();
+         i++) {
+      assertDoesNotThrow(() -> misuseMonitor.notifyEvent(user.getEmail(), event),
+          String.format("Exception thrown before %d attempts, under limit of %d!", i,
+              emailVerificationMisuseHandler.getHardThreshold()));
     }
+
+    assertThrows(SegueResourceMisuseException.class, () -> misuseMonitor.notifyEvent(user.getEmail(), event),
+        String.format("SegueResourceMisuseException not thrown after %d attempts, over limit of %d!",
```

Comment:
Same here. Please, have another look at this whole class, as there are a few messages that don't look right.

Reply:
Yeah, I think I might just redo those messages in general... 🤔